File: .agents/skills/release/cdr-updater/src/com/jsi/cdr/updater/Policy.java
package com.jsi.cdr.updater;

import java.util.Map;
import java.util.Set;
import java.util.stream.Collectors;

final class Policy {
    private Policy() {}

    static boolean shouldDeleteLocal(String path, Set<String> managedPaths) {
        String rel = Fs.posix(path);
        if (isSelf(rel) || PackPaths.protectedLocal(rel)) {
            return false;
        }
        return managed(managedPaths).contains(rel);
    }

    static boolean shouldOverwriteLocal(
            String path,
            String side,
            String localSha,
            String remoteSha,
            Map<String, String> syncedHashes
    ) {
        return shouldOverwriteLocal(path, side, localSha, remoteSha, syncedHashes, false, null, false);
    }

    static boolean shouldOverwriteLocal(
            String path,
            String side,
            String localSha,
            String remoteSha,
            Map<String, String> syncedHashes,
            boolean overlay
    ) {
        return shouldOverwriteLocal(path, side, localSha, remoteSha, syncedHashes, overlay, null, false);
    }

    static boolean shouldOverwriteLocal(
            String path,
            String side,
            String localSha,
            String remoteSha,
            Map<String, String> syncedHashes,
            boolean overlay,
            Boolean localHasManagedTag
    ) {
        return shouldOverwriteLocal(path, side, localSha, remoteSha, syncedHashes, overlay, localHasManagedTag, false);
    }

    static boolean shouldOverwriteLocal(
            String path,
            String side,
            String localSha,
            String remoteSha,
            Map<String, String> syncedHashes,
            boolean overlay,
            Boolean localHasManagedTag,
            boolean keepLocal
    ) {
        String rel = Fs.posix(path);
        if (isSelf(rel) || remoteSha == null || remoteSha.isBlank()) {
            return false;
        }
        if (localSha == null) {
            return true;
        }
        if (localSha.equals(remoteSha)) {
            return false;
        }
        if (PackPaths.taggedTemplate(rel)) {
            if (keepLocal) {
                return false;
            }
            if (!Boolean.TRUE.equals(localHasManagedTag)) {
                return true;
            }
            String previous = syncedHashes == null ? null : syncedHashes.get(rel);
            return previous != null && localSha.equals(previous);
        }
        if (overlay && !PackPaths.protectedLocal(rel)) {
            return true;
        }
        if (PackPaths.protectedLocal(rel)) {
            return false;
        }
        if (!"server".equals(side)) {
            return true;
        }
        String previous = syncedHashes == null ? null : syncedHashes.get(rel);
        if (previous == null) {
            return false;
        }
        return localSha.equals(previous);
    }

    private static boolean isSelf(String rel) {
        return "mods/cdr-updater.jar".equals(rel);
    }

    private static Set<String> managed(Set<String> managedPaths) {
        return managedPaths.stream().map(Fs::posix).collect(Collectors.toSet());
    }
}

File: .agents/skills/release/cdr-updater-go/server/policy.go
package main

func shouldDeleteLocal(path string, managed map[string]struct{}) bool {
	rel := posix(path)
	if protectedLocal(rel) {
		return false
	}
	_, ok := managed[rel]
	return ok
}

func shouldOverwriteLocal(path, side, localSha, remoteSha string, synced map[string]string, overlay bool, localHasTag *bool, keepLocal bool) bool {
	rel := posix(path)
	if remoteSha == "" {
		return false
	}
	if localSha == "" {
		return true
	}
	if localSha == remoteSha {
		return false
	}
	if taggedTemplate(rel) {
		if keepLocal {
			return false
		}
		if localHasTag == nil || !*localHasTag {
			return true
		}
		previous, ok := synced[rel]
		return ok && localSha == previous
	}
	if overlay && !protectedLocal(rel) {
		return true
	}
	if protectedLocal(rel) {
		return false
	}
	if side != "server" {
		return true
	}
	previous, ok := synced[rel]
	if !ok {
		return false
	}
	return localSha == previous
}
